src/isomorphic/classic/types/ReactPropTypes.js

```diff
@@ -208,11 +210,13 @@ function createInstanceTypeChecker(expectedClass) {
 
 function createEnumTypeChecker(expectedValues) {
   if (!Array.isArray(expectedValues)) {
-    return createChainableTypeChecker(function() {
+    var error = createChainableTypeChecker(function() {
```

Comment:
Since this is the example I keep using, let's work through it.

1. We put `if (__DEV__) { warning(false, 'Some message here') }
2. Continue returning a chainable type checker (because isRequired can still be added).
3. We can perhaps just make it so that we pass it an empty function in or we keep it returning an error but we change that message to make it clear that this prop failure can be ignored if you are not the component author. I'm leaning towards the former right this second (because it's not terribly useful to have to work past a failure you aren't responsible for).